content/algorithms/dag/index.md

```diff
@@ -0,0 +1,362 @@
+(for example, when washing clothes, the washing machine must finish before we put the clothes in the dryer).
+Then, a topological sort gives an order in which to perform the jobs.
+
+A closely related application of topological sorting algorithms
+was first studied in the early 1960s in the context of the
+[PERT technique](https://en.wikipedia.org/wiki/Program_evaluation_and_review_technique)
+for scheduling in project management.
+In this application, the vertices of a graph represent the milestones of a project,
+and the edges represent tasks that must be performed between one milestone and another.
+Topological sorting forms the basis of linear-time algorithms for finding
+the critical path of the project, a sequence of milestones and tasks that controls
+the length of the overall project schedule.
+
+In computer science, applications of this type arise in instruction scheduling,
+ordering of formula cell evaluation when recomputing formula values in spreadsheets,
+logic synthesis, determining the order of compilation tasks to perform in makefiles,
+data serialization, and resolving symbol dependencies in linkers.
+It is also used to decide in which order to load tables with foreign keys in databases.
+
+### Definition
+
+A topological sort of a directed acyclic graph $G = (V, E)$ is a linear ordering of all its vertices
+such that if $G$ contains an edge $(u, v)$, then $u$ appears before $v$ in the ordering.
+
+It is worth noting that if the graph contains a cycle, then no linear ordering is possible.
+
+It is useful to view a topological sort of a graph as an ordering of its vertices
+along a horizontal line so that all directed edges go from left to right.
+
+### Kahn's algorithm
+
+NetworkX uses Kahn's algorithm to perform topological sorting.
+We will introduce it briefly here.
+
+First, find a list of "start nodes" which have no incoming edges and insert them into a set S;
+at least one such node must exist in a non-empty acyclic graph. Then:
+
+```
+L <- Empty list that will contain the sorted elements
+S <- Set of all nodes with no incoming edge
+
+while S is not empty do
+    remove a node N from S
+    add N to L
+    for each node M with an edge E from N to M do
+        remove edge E from the graph
+        if M has no other incoming edges then
+            insert M into S
+
+if graph has edges then
+    return error  # graph has at least one cycle
+else 
+    return L  # a topologically sorted order
+```
+
+### NetworkX implementation
+
+Finally, let's take a look at how the topological sorting is implemented in NetworkX.
+
+We can see that Kahn's algorithm _stratifies_ the graph such that each level contains all the nodes
+whose dependencies have been satisfied by the nodes in a previous level.
+What this translates to is, take all the nodes in the DAG that don't have any dependencies and put them in list.
+Then "remove" those nodes from the DAG, and repeat the process, creating a new list at each step.
```

Comment:
```suggestion
In other words, the general procedure looks something like:
 - Take all the nodes in the DAG that don't have any dependencies and put them in list.
 - "Remove" those nodes from the DAG.
 - Repeat the process, creating a new list at each step.
```

Just a suggestion to make the procedure more digestible. You could consider an enumerated list too, or feel free to ignore entirely!